apps/vs-code-designer/src/app/utils/taskUtils.ts

```diff
@@ -61,3 +62,27 @@ export async function executeIfNotActive(task: Task): Promise<void> {
     await codeTasks.executeTask(task);
   }
 }
+
+/**
+ * Unzips the contents of a Logic App project into a target directory.
+ * This function uses the AdmZip library to handle the unzipping operation.
+ *
+ * @param {Buffer} zipContent - The buffer containing the compressed Logic App project.
+ * @param {string} targetDirectory - The path of the directory where the unzipped files will be stored.
+ * @returns {Promise<void>} - A Promise that resolves when the unzipping process is complete, or rejects with an error.
+ *
+ * @throws Will throw an error if the unzipping process fails.
+ */
+export async function unzipLogicAppArtifacts(zipContent: Buffer, targetDirectory: string): Promise<void> {
+  try {
+    // Initialize a new AdmZip object with the provided zip content
+    const zip = new AdmZip(zipContent);
+
+    // Extract all the files from the zip content to the target directory
+    // The second parameter set to 'true' indicates that it will overwrite existing files in the target directory
+    zip.extractAllTo(targetDirectory, true);
+  } catch (error) {
+    window.showErrorMessage('Failed to unzip Logic App:', error);
```

Comment:
We don't capitalize "logic app" when referring to the resource instance:

```suggestion
    window.showErrorMessage('Failed to unzip logic app: ', error);
```